File: leet-code/countRangeSum.go
package main

func countRangeSum(nums []int, lower int, upper int) int {
	count := 0
	for i := range nums {
		sum := nums[i]
		if sum >= lower && sum <= upper {
			count++
		}
		for j := i + 1; j < len(nums); j++ {
			sum += nums[j]
			if sum >= lower && sum <= upper {
				count++
			}
		}
	}
	return count
}

File: leet-code/countRangeSum.js
/**暴力循环
 * @param {number[]} nums
 * @param {number} lower
 * @param {number} upper
 * @return {number}
 */
var countRangeSum = function (nums, lower, upper) {
  let count = 0
  for (let i = 0; i < nums.length; i++) {
    let sum = nums[i]
    if (nums[i] >= lower && nums[i] <= upper) {
      count++
    }
    for (let j = i + 1; j < nums.length; j++) {
      sum += nums[j]
      if (sum >= lower && sum <= upper) {
        count++
      }
    }
  }
  return count
}

console.log(countRangeSum([0, 0], 0, 0))
